sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java

```diff
@@ -68,6 +68,9 @@ public ComponentDto selectOrFailByUuid(DbSession session, String uuid) {
   }
 
   public List<ComponentDto> selectByQuery(DbSession session, ComponentQuery query, int offset, int limit) {
+    if (query.getComponentIds() != null && query.getComponentIds().isEmpty()) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'getComponentIds' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)